### ISSUE TITLE
AttributeError: module 'boto3.s3.inject' has no attribute 'inject_s3_transfer_methods'

### ISSUE THREAD

**Issue by dangitall** [pyinstaller/pyinstaller#2450]
When running multiple downloads or uploads to S3 in parallel, I get this error when running an app packaged with pyinstaller. The original python code runs without an issue.

I see this issue with Python 3.5.2 and both pyinstaller 3.2.1 and 3.3.dev0+11f2e7b. Here is a simple test case that reproduces the code for me 100% of the time, but I'm sorry that I don't have a public S3 bucket to point you to. Hopefully that is something that can be worked around.

```
import sys
import boto3
import concurrent.futures
import encodings.idna


def dl(bucket, key):
    boto3.session.Session().client('s3').download_file(bucket, key, "/tmp/x")


exc = concurrent.futures.ThreadPoolExecutor()

res = []
for i in range(100):
    res.append(exc.submit(dl, sys.argv[1], sys.argv[2]))

[r.result() for r in res]
```

My command line is `pyinstaller --clean --onedir --noupx --debug boto3test.spec` and here is my spec file:

```
# -*- mode: python -*-

# work-around a bug in hook-botocore
from PyInstaller.utils.hooks import is_module_satisfies
import PyInstaller.compat
PyInstaller.compat.is_module_satisfies = is_module_satisfies


block_cipher = None


a = Analysis(['boto3test.py'],
             pathex=['/Users/dan/hack/pyinstaller-boto3-test'],
             binaries=[],
             datas=[],
             hiddenimports=[],
             hookspath=[],
             runtime_hooks=[],
             excludes=[],
             win_no_prefer_redirects=False,
             win_private_assemblies=False,
             cipher=block_cipher)
pyz = PYZ(a.pure, a.zipped_data,
             cipher=block_cipher)
exe = EXE(pyz,
          a.scripts,
          exclude_binaries=True,
          name='boto3test',
          debug=False,
          strip=False,
          upx=False,
          console=True )
coll = COLLECT(exe,
               a.binaries,
               a.zipfiles,
               a.datas,
               strip=False,
               upx=False,
               name='boto3test')
```

Packages installed:
```boto3==1.4.4
botocore==1.5.9
docutils==0.13.1
jmespath==0.9.1
PyInstaller==3.3.dev0+11f2e7b
python-dateutil==2.6.0
s3transfer==0.1.10
six==1.10.0```

**Comment by xoviat** [comment]
Can you apply #2377?

```
git clone https://github.com/pyinstaller/pyinstaller.git
cd pyinstaller
git pull origin +refs/pull/2377/merge
pip install --force-reinstall .
```

Note: executables generated with unmerged pull requests may be unstable and should not be distributed to users without prior verification.

**Comment by dangitall** [comment]
Yes, that seems to have fixed it. Thanks!

**Comment by xoviat** [comment]
Please close the issue then.

**Comment by dangitall** [comment]
Shouldn't it remain open until that PR is merged?

**Comment by xoviat** [comment]
No, because this issue is actually a duplicate of #2371.

**Comment by dangitall** [comment]
Fair enough!
